src/components/Header/Header.astro

```diff
@@ -19,6 +22,12 @@ const algoliaVars = {
   width="0"
   style="display:none;visibility:hidden"></iframe>
 <!-- End Google Tag Manager (noscript) -->
-<NavBar path={path} productsNav={productsNav} subProductsNav={subProductsNav} algoliaVars={algoliaVars} client:idle />
+<NavBar
+  path={path}
+  productsNav={productsNav}
+  subProductsNav={subProductsNav}
+  algoliaVars={algoliaVars}
+  client:only="react"
```

Comment:
This was added because the search component uses react markdown which uses DOM apis like `window` and `document`.